src/Plugin/Field/FieldWidget/ColorSchemeWidget.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_color_scheme\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Extension\ThemeExtensionList;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Theme\ThemeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'Color Scheme' widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_color_scheme_widget",
+ *   label = @Translation("Color Scheme widget"),
+ *   field_types = {
+ *     "oe_color_scheme"
+ *   }
+ * )
+ */
+class ColorSchemeWidget extends WidgetBase {
+
+  /**
+   * The theme manager.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected $themeManager;
+
+  /**
+   * The theme extension list.
+   *
+   * @var \Drupal\Core\Extension\ThemeExtensionList
+   */
+  protected $themeExtensionList;
+
+  /**
+   * Constructs a ColorSchemeWidget object.
+   *
+   * @param string $plugin_id
+   *   The plugin_id for the widget.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the widget is associated.
+   * @param array $settings
+   *   The widget settings.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   * @param \Drupal\Core\Theme\ThemeManagerInterface $theme_manager
+   *   The theme manager.
+   * @param \Drupal\Core\Extension\ThemeExtensionList $theme_extension_list
+   *   The theme extension list.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, ThemeManagerInterface $theme_manager, ThemeExtensionList $theme_extension_list) {
```

Comment:
IMO we should use property promotion in the constructor to simplify the code (it also eliminates two previous comments). There is no official recommendation for it, but Drupal core already [does it](https://github.com/drupal/drupal/blob/10.3.x/core/modules/field_ui/src/Access/FieldReuseAccessCheck.php).

The downside is that we mix camelCase with snake_case, but CS should ignore it.